src/main/java/Main.java

```diff
@@ -8,6 +9,92 @@ public static void main(String[] args) {
 
     public void run(Scanner scanner) {
+        ArrayList<Integer> numbers = inputNumbers(scanner);
+        printReversedNumbers(numbers);
+        printResultOfAdding(numbers);
+        printTheSmallestNumber(numbers);
+        printTheBiggestNumber(numbers);
+
         // napisz swój program tutaj. Do wczytywania danych użyj przekazanego w parametrze scannera
     }
+
+    private void printTheSmallestNumber(ArrayList<Integer> numbers) {
+        if (!numbers.isEmpty()) {
+            int smallestNumber = numbers.get(0);
+            for (Integer number : numbers) {
+                if (number < smallestNumber) {
+                    smallestNumber = number;
+                }
+            }
+            System.out.println("Najmniejsza liczba w liście to " + smallestNumber);
+        }  else {
+            System.out.println("Lista jest pusta");
+        }
+    }
+
+    private void printTheBiggestNumber(ArrayList<Integer> numbers) {
+        if (!numbers.isEmpty()) {
+            int biggestNumber = numbers.get(0);
+            for (Integer number : numbers) {
+                if (number > biggestNumber) {
+                    biggestNumber = number;
+                }
+            }
+            System.out.println("Największa liczba w liście to " + biggestNumber);
+        }  else {
+            System.out.println("Lista jest pusta");
+        }
+    }
+
+    private void printResultOfAdding(ArrayList<Integer> numbers) {
+        if (!numbers.isEmpty()) {
+            int sum = 0;
+            StringBuilder sb = new StringBuilder();
+            for (Integer number : numbers) {
+                sb.append(number);
+                sb.append(" + ");
+                sum += number;
+            }
+            sb.replace(sb.length() - 2, sb.length(), "= ");
+            sb.append(sum);
+            System.out.println(sb);
+        } else {
+            System.out.println("Lista jest pusta");
+        }
+
+    }
+
+    private void printReversedNumbers(ArrayList<Integer> numbers) {
+        if (!numbers.isEmpty()) {
+            StringBuilder sb = new StringBuilder();
+            for (int i = numbers.size() - 1; i >= 0; i--) {
+                sb.append(numbers.get(i));
+                sb.append(", ");
+            }
+            sb.replace(sb.length() - 2, sb.length(), "");
+            System.out.println(sb);
+        } else {
+            System.out.println("Lista jest pusta");
+        }
+
+    }
+
+    private static ArrayList<Integer> inputNumbers(Scanner scanner) {
+        ArrayList<Integer> numbers = new ArrayList<>();
+        int status = 1;
+        do {
+            int number = scanner.nextInt();
+            if (number < 0) {
+                status = number;
+                break;
+            }
+            numbers.add(number);
+        } while (status >= 0);
+        if (numbers.isEmpty()) {
+            System.out.println("Lista jest pusta, brak możliwości wykonania dodatkowych operacji.");
+            System.exit(1);
```

Comment:
Stosowanie System.exit() jest najczęściej błędne, więcej info tutaj https://youtu.be/HsrEsJWwCSM

W tym przypadku niech już zostanie, zwracaj na to uwagę w przyszłości.